components/espcoredump/test/test_core_dump.c

```diff
@@ -96,10 +98,43 @@ void failed_assert_task(void *pvParameter)
     fflush(stdout);
 }
 
+void spi_flash_disable_interrupts_caches_and_other_cpu();
+
+void IRAM_ATTR crash_from_ram_no_flash_cache()
+{
+    spi_flash_disable_interrupts_caches_and_other_cpu();
+
+    printf("Crashed by calling printf while cache is disabled\n");
+}
+
+void crash_from_no_flash_cache()
+{
+    printf("Task 'crash_from_no_flash_cache' start.\n");
+    while (1) {
+        vTaskDelay(1000 / portTICK_RATE_MS);
+        printf("Task 'crash_from_no_flash_cache' run.\n");
+        crash_from_ram_no_flash_cache();
+    }
+    fflush(stdout);
+}
+
+void starve_idle()
```

Comment:
It helped see issues with the panic handler, for instance, of the elf sha256 sum being weird in this case.  You're probably right in that this probably more aptly belongs inside a test case for the panic handler, and not the coredump component.

In our local repository, however, I do have branch code within the coredump component dependent on if flash cache is enabled.  (I have a custom backend for the coredump routine which dumps the information into a filesystem that isn't supported within esp-idf - I've been poking at https://github.com/ARMmbed/littlefs.)  Hence why I came across some of the issues present in some of the wrapper code around running coredumps as I was debugging my coredump implementation and hitting some crashes within it.  (For instance, I had a crash in my coredump code when cache flash was disabled, in which I hit the bug fixed in #3661.  That was being tested by a test case similar to this, although, I have it implemented as a test within our console code instead of this test case specifically, but the test is the same.)